app/collectors/instance.scala

```diff
@@ -2,41 +2,43 @@ package collectors
 
 import org.joda.time.DateTime
 
-import scala.collection.JavaConversions._
+import scala.jdk.CollectionConverters._
 import utils.{Logging, PaginatedAWSRequest}
 import java.net.InetAddress
 
-import play.api.libs.json.Json
+import play.api.libs.json.{Json, Reads}
 import play.api.mvc.Call
-import controllers.routes
+import controllers.{Prism, routes}
 
 import scala.language.postfixOps
-import com.amazonaws.services.ec2.AmazonEC2ClientBuilder
+import com.amazonaws.services.ec2.{AmazonEC2, AmazonEC2ClientBuilder}
 import com.amazonaws.services.ec2.model.{DescribeInstancesRequest, Instance => AWSInstance, Reservation => AWSReservation}
 import agent._
+
 import scala.concurrent.duration._
+import scala.util.matching.Regex
 
 
-object InstanceCollectorSet extends CollectorSet[Instance](ResourceType("instance", 15 minutes, 1 minute)) {
+class InstanceCollectorSet(accounts: Accounts, prism: Prism) extends CollectorSet[Instance](ResourceType("instance", 15 minutes, 1 minute), accounts) {
   val lookupCollector: PartialFunction[Origin, Collector[Instance]] = {
-    case json:JsonOrigin => JsonInstanceCollector(json, resource)
-    case amazon:AmazonOrigin => AWSInstanceCollector(amazon, resource)
+//    case json:JsonOrigin => JsonInstanceCollector(json, resource)
+    case amazon:AmazonOrigin => AWSInstanceCollector(amazon, resource, prism)
   }
 }
 
-case class JsonInstanceCollector(origin:JsonOrigin, resource:ResourceType) extends JsonCollector[Instance] {
-  import jsonimplicits.joda.dateTimeReads
-  import jsonimplicits.model._
-  implicit val addressReads = Json.reads[Address]
-  implicit val instanceSpecificationReads = Json.reads[InstanceSpecification]
-  implicit val managementEndpointReads = Json.reads[ManagementEndpoint]
-  implicit val instanceReads = Json.reads[Instance]
-  def crawl: Iterable[Instance] = crawlJson
-}
+//case class JsonInstanceCollector(origin:JsonOrigin, resource:ResourceType) extends JsonCollector[Instance] {
```

Comment:
perhaps this can be removed completely? @sihil might be able to confirm that this is no longer used now that we are 100% in aws

Reply:
🔨